docs/user-guide.md

```diff
@@ -75,6 +76,11 @@ spec:
   mainApplicationFile: local:///opt/spark/examples/jars/spark-examples_2.12-3.0.0.jar
 ```
 
+### Specifying Deployment Mode
+
+A `SparkApplication` should set `.spec.deployMode` to `cluster`, as `client` is not currently implemented. The driver pod will then run `spark-submit` in `client` mode internally to run the driver program. Additional details of how `SparkApplication`s are run can be found in the [design documentation](design.md#architecture).
```

Comment:
@psschwei : Shouldn't the field be `.spec.mode` instead of `.spec.deployMode` ?
Looking at examples here: https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/user-guide.md#writing-a-sparkapplication-spec

Reply:
It's been over two years since I've used the spark operator, so I really couldn't say.